src/Microsoft.OData.Core/Evaluation/ODataConventionalEntityMetadataBuilder.cs

```diff
@@ -355,6 +355,9 @@ internal override IEnumerable<ODataFunction> GetFunctions()
             return ODataUtilsInternal.ConcatEnumerables(this.ResourceMetadataContext.Resource.NonComputedFunctions, this.MissingOperationGenerator.GetComputedFunctions());
         }
 
+        // mikep todo: why does MarkNestedResourceInfoProcessed and GetNextUprocessedNavigationLink exist here with 
+        // apparently same implementation as in base class?

```

Comment:
From the implementation, 

if (structuredType.IsODataEntityTypeKind())
{
    resourceState.MetadataBuilder = new ODataConventionalEntityMetadataBuilder(resourceMetadataContext, this, uriBuilder);
}
else
{
     resourceState.MetadataBuilder = new ODataConventionalResourceMetadataBuilder(resourceMetadataContext, this, uriBuilder);
}

The implementation in ODataConventionalResourceMetadataBuilder should be removed.

Reply:
Actually, I'm guessing that the code was added to ODataConventionalResourceMetadataBuilder when it started supporting nav properties on complex types.  The implementation on the derived ODataConventionalEntityMetadataBuilder is now redundant and should be removed, in favor of the (identical) implementation in ODataConventionalResourceMetadataBuilder.

---
In reply to: [258642599](https://github.com/OData/odata.net/pull/1204#discussion_r258642599) [](ancestors = 258642599)